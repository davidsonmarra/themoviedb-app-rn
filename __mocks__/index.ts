import MovieDTO from '../src/@types/movies-dto';

export const mockCardMovie: MovieDTO = {
  id: 1,
  title: 'Movie Title',
  overview: 'Movie Overview',
  poster_path: '7qOSKoOAPgemYhBwbJgBWcCxPWZ.jpg',
  release_date: '2021-01-01',
  backdrop_path: 'path/to/image',
  vote_average: 10,
  genre_ids: [1, 2],
};

export const mockListMovie: MovieDTO[] = [
  {
    id: 1,
    title: 'Movie Title',
    overview: 'Movie Overview',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 10,
    genre_ids: [1, 2],
  },
  {
    id: 2,
    title: 'Movie Title 2',
    overview: 'Movie Overview 2',
    poster_path: 'path/to/image',
    release_date: '2021-01-01',
    backdrop_path: 'path/to/image',
    vote_average: 9,
    genre_ids: [1, 2],
  },
];
