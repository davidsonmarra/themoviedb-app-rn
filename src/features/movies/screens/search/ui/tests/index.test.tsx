import React from 'react';
import SearchContainer from '..';
import {render, screen} from '@testing-library/react-native';
import {mockListMovie} from '../../../../../../../__mocks__';

const mockGoBack = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({goBack: mockGoBack}),
}));

const containerInstance = (props: any) =>
  render(<SearchContainer {...props} />);

describe('SearchContainer', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const container = containerInstance({
      search: '',
      searchedMoviesData: mockListMovie,
      loadingFetchSearchMovies: false,
      onEndReached: jest.fn(),
      setSearch: jest.fn(),
      onSearchMovies: jest.fn(),
      onPressMovie: jest.fn(),
    }).toJSON();
    expect(container).toMatchSnapshot();
  });

  it('should render movie details', () => {
    containerInstance({
      search: '',
      searchedMoviesData: mockListMovie,
      loadingFetchSearchMovies: false,
      onEndReached: jest.fn(),
      setSearch: jest.fn(),
      onSearchMovies: jest.fn(),
      onPressMovie: jest.fn(),
    });
    const title = screen.getByText(mockListMovie[0].title);
    expect(title).toBeTruthy();
  });

  it('should call onSearchMovies when search button is pressed', () => {
    const onSearchMovies = jest.fn();
    containerInstance({
      search: '',
      searchedMoviesData: mockListMovie,
      loadingFetchSearchMovies: false,
      onEndReached: jest.fn(),
      setSearch: jest.fn(),
      onSearchMovies,
      onPressMovie: jest.fn(),
    });
    const button = screen.getByTestId('search-button');
    button.props.onClick();
    expect(onSearchMovies).toHaveBeenCalled();
  });

  it('should call onPressMovie when movie is pressed', () => {
    const onPressMovie = jest.fn();
    containerInstance({
      search: '',
      searchedMoviesData: mockListMovie,
      loadingFetchSearchMovies: false,
      onEndReached: jest.fn(),
      setSearch: jest.fn(),
      onSearchMovies: jest.fn(),
      onPressMovie,
    });
    const movie = screen.queryAllByTestId('movie-card');
    movie[0].props.onClick();
    expect(onPressMovie).toHaveBeenCalled();
  });
});
