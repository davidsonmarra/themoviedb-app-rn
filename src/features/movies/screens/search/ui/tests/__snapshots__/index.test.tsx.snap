// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should render correctly 1`] = `
<View>
  <View
    style={
      {
        "backgroundColor": "#01B4A4",
        "flexDirection": "row",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="back-button"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="24"
        bbWidth="24"
        fill="#FFFFFF"
        focusable={false}
        height="24"
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        vbHeight={960}
        vbWidth={960}
        width="24"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M359-242L120-481l239-239 43 43-166 166h604v60H236l166 166-43 43z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
          },
          {
            "flex": 1,
            "textAlign": "center",
          },
        ]
      }
    >
      Procurar Filme
    </Text>
    <View
      style={
        {
          "width": 12,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": 24,
        }
      }
      testID="right-buton-icon"
    />
  </View>
  <View
    style={
      {
        "paddingHorizontal": 16,
        "paddingVertical": 24,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "borderColor": "#595959",
          "borderRadius": 5,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        onChangeText={[MockFunction]}
        style={
          {
            "flex": 1,
            "height": 40,
          }
        }
        testID="input"
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="search-button"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="#01B4A4"
          focusable={false}
          height="24"
          meetOrSlice={0}
          minX={0}
          minY={-960}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={960}
          vbWidth={960}
          width="24"
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4278301860,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M784-120L532-372q-30 24-69 38t-83 14q-109 0-184.5-75.5T120-580q0-109 75.5-184.5T380-840q109 0 184.5 75.5T640-580q0 44-14 83t-38 69l252 252-56 56zM380-400q75 0 127.5-52.5T560-580q0-75-52.5-127.5T380-760q-75 0-127.5 52.5T200-580q0 75 52.5 127.5T380-400z"
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      ListFooterComponent={[Function]}
      contentContainerStyle={
        {
          "paddingBottom": 100,
          "paddingVertical": 16,
        }
      }
      data={
        [
          {
            "backdrop_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
            "genre_ids": [
              1,
              2,
            ],
            "id": 1,
            "overview": "Movie Overview",
            "poster_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
            "release_date": "2021-01-01",
            "title": "Movie Title",
            "vote_average": 10,
          },
          {
            "backdrop_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
            "genre_ids": [
              1,
              2,
            ],
            "id": 2,
            "overview": "Movie Overview 2",
            "poster_path": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
            "release_date": "2021-01-01",
            "title": "Movie Title 2",
            "vote_average": 9,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      keyboardDismissMode="on-drag"
      onContentSizeChange={[Function]}
      onEndReached={[MockFunction]}
      onEndReachedThreshold={0.5}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={16}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      testID="movies-flatlist"
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "#CCCCCC",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 16,
                "opacity": 1,
              }
            }
            testID="movie-card"
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 8,
                  "borderTopLeftRadius": 8,
                  "height": 141,
                  "width": 94,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "padding": 12,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#121212",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Movie Title
              </Text>
              <Text
                style={
                  {
                    "color": "#808080",
                    "fontSize": 14,
                  }
                }
              >
                31/12/2020
              </Text>
              <Text
                numberOfLines={3}
                style={
                  {
                    "color": "#222222",
                    "fontSize": 14,
                    "marginTop": 8,
                  }
                }
              >
                Movie Overview
              </Text>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "borderColor": "#CCCCCC",
                "borderRadius": 8,
                "borderWidth": 1,
                "flexDirection": "row",
                "marginBottom": 16,
                "opacity": 1,
              }
            }
            testID="movie-card"
          >
            <Image
              source={
                {
                  "uri": "https://image.tmdb.org/t/p/w188_and_h282_bestv2/https://image.tmdb.org/t/p/w188_and_h282_bestv2/wNAhuOZ3Zf84jCIlrcI6JhgmY5q.jpg",
                }
              }
              style={
                {
                  "borderBottomLeftRadius": 8,
                  "borderTopLeftRadius": 8,
                  "height": 141,
                  "width": 94,
                }
              }
            />
            <View
              style={
                {
                  "flex": 1,
                  "padding": 12,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#121212",
                    "fontSize": 16,
                    "fontWeight": "bold",
                  }
                }
              >
                Movie Title 2
              </Text>
              <Text
                style={
                  {
                    "color": "#808080",
                    "fontSize": 14,
                  }
                }
              >
                31/12/2020
              </Text>
              <Text
                numberOfLines={3}
                style={
                  {
                    "color": "#222222",
                    "fontSize": 14,
                    "marginTop": 8,
                  }
                }
              >
                Movie Overview 2
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              {
                "height": 100,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
